spec/coupon_code_spec.rb

```diff
@@ -1,56 +1,76 @@
+# frozen_string_literal: true
+
 require 'spec_helper'
 
 RSpec.describe CouponCode do
-
   describe '.generate' do
     subject { described_class.generate }
+
     it { is_expected.not_to be_nil }
-    it { is_expected.to match(/^[0-9A-Z-]+$/) }
+    it { is_expected.to match(/^[2-9A-Z-]+$/) }
     it { is_expected.to match(/^\w{4}-\w{4}-\w{4}$/) }
     it 'generates a different code' do
       code2 = described_class.generate
       is_expected.not_to eq(code2)
     end
 
-    context '2 parts' do
+    context 'when 2 parts' do
       subject { described_class.generate(parts: 2) }
       it { is_expected.to match(/^\w{4}-\w{4}$/) }
```

Comment:
```suggestion
      subject { described_class.generate(parts: 2) }
      
      it { is_expected.to match(/^\w{4}-\w{4}$/) }
```

Reply:
Done. There is no rubocop/rubocop-rspec setup for this gem currently, so we should handle formatting manually.